pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -2576,7 +2576,7 @@ with the restriction that the logger needs to be passed into the constructor.
 //ClassOrInterfaceBodyDeclaration[FieldDeclaration//ClassOrInterfaceType[@Image='Log']
  and
  not(FieldDeclaration[@Final='true'][@Static='true'][@Private='true'][.//VariableDeclaratorId[@Image=$staticLoggerName]]
- //ArgumentList//ClassOrInterfaceType/@Image = ancestor::ClassOrInterfaceDeclaration/@Image)
+ //ArgumentList//ClassOrInterfaceType/@Image = (ancestor::ClassOrInterfaceDeclaration/@Image) or (ancestor::EnumDeclaration/@Image) )
```

Comment:
this will whitelist any logger defined in an enum. You are asking for:

* `//ArgumentList//ClassOrInterfaceType/@Image = (ancestor::ClassOrInterfaceDeclaration/@Image)`
OR
* `ancestor::EnumDeclaration/@Image`

Try adding a test case:

```
public enum Foo {
    TEST;
    private static final Log LOG = LogFactory.getLog(Bar.class);
}
```

and check that 1 issue is reported